Add select values provider service documentation (#570)
sulu/sulu-docs: cookbook/select-values-provider.rst

Create a select values provider
===============================

> An example :doc:`../reference/content-types/single_select` or :doc:`../reference/content-types/select` values provider could look like the following:

Reviewer: ```
Sulu allows to set the value of a ``param`` of a property with the type ``expression`` via the Symfony expression language (https://symfony.com/doc/current/components/expression_language.html). 
The makes it possible to display values returned by a service in the case of the  :doc:`../reference/content-types/select` and :doc:`../reference/content-types/single_select` content-type.

.. note::

    Be aware that the provided expression is only evaluated during the initial request to the administration interface.
    If you want to provide a selection for your custom entity, you should configure the ``single_selection`` field-type
    as described in :doc:`../../book/extend-admin`.

A service for returning the values for these content types could look something like this:
```

.. code-block:: php

    <?php

    namespace App\Content\Select;

    class EventTypeSelect {
        /**
         * @return array<int, array{name: string, title: string}>
         */
        public function getValues(): array
        {
            return [
                [
                    'name' => 'concert',

>                     'title' => 'Concert',

Reviewer: can we return a localized title?

Author: Not sure I never tried it and it seems the expression language does currently not support getting the `locale`.

                ],
                [
                    'name' => 'movie',
                    'title' => 'Movie',
                ],
            ];
        }

        /**
         * Optional default value for a single select.
         */

>         public function getDefaultValue(): string

Reviewer: ```suggestion
        public function getSingleSelectDefaultValue(): string
```

        {
            return 'concert';
        }

        /**
         * Optional default value for a multi select.
         *
         * @return array<int, array{name: string}>
         */

>         public function getDefaultValueMultiSelect(): array

Reviewer: ```suggestion
        public function getMultiSelectDefaultValue(): array
```

        {
            return [
                ['name' => 'concert'],
            ];
        }
    }

> The service need to be available as public service so the following need to be added to the ``config/services.yaml``:

Reviewer: ```
The service need to be marked as public (https://symfony.com/doc/current/service_container/alias_private.html). This is possible by adding the following lines to the ``config/services.yaml``:
```

.. code-block:: yaml

    App\Content\Select\:
        resource: '../src/Content/Select/'
        public: true

> Then you can use the value provider the following way in your pages template:

Reviewer: ```suggestion
When your service is marked as public, you can access it in your template like this:
```
